### PR TITLE
fix: add required fields existance checks in loader

### PR DESCRIPTION
**Pull request by DawoudSheraz**
### Description
- Add checks in CSV loader to validate the data existence for required fields for minimal ingestion based on course type

### CODE REVIEW

File: course_discovery/apps/course_metadata/data_loaders/csv_loader.py [hunk 1]
```diff
@@ -30,6 +31,18 @@ class CSVDataLoader(AbstractDataLoader):
...
     ]
+    # list of data fields (present as CSV columns) that should be present in each row
+    BASE_REQUIRED_DATA_FIELDS = [
+        'title', 'number', 'image', 'short_description', 'long_description', 'what_will_you_learn', 'course_level',
+        'primary_subject', 'verified_price', 'syllabus', 'frequently_asked_questions', 'publish_date', 'start_date',
+        'start_time', 'end_date', 'end_time', 'course_pacing', 'minimum_effort', 'maximum_effort', 'length',
+        'content_language', 'transcript_language'
+    ]
+
+    EXECUTIVE_EDUCATION_REQUIRED_FIELDS = BASE_REQUIRED_DATA_FIELDS + [
```
Comment (azanbinzahid) [inline course_discovery/apps/course_metadata/data_loaders/csv_loader.py:42]: Can be renamed to `AdditionalMetadata`?

File: course_discovery/apps/course_metadata/data_loaders/csv_loader.py [hunk 2]
```diff
@@ -392,7 +431,10 @@ def get_subject_slugs(self, *subjects):
                 sub_obj = Subject.objects.get(translations__name=subject, translations__language_code='en')
                 subject_slugs.append(sub_obj.slug)
             except Subject.DoesNotExist:
-                logger.exception("Unable to locate subject {} in the database. Skipping subject association")
+                logger.exception("Unable to locate subject {} in the database. Skipping subject association".format(
```
Comment (azanbinzahid) [inline course_discovery/apps/course_metadata/data_loaders/csv_loader.py:434]: `f-string` may be?

**Review by azanbinzahid — APPROVED** [review]

File: course_discovery/apps/course_metadata/data_loaders/csv_loader.py (hunk 2 above)
Comment (DawoudSheraz) [reply course_discovery/apps/course_metadata/data_loaders/csv_loader.py:434]: Using f-strings in log statement causes pylint quality failure (https://pylint.pycqa.org/en/latest/messages/warning/logging-fstring-interpolation.html)

File: course_discovery/apps/course_metadata/data_loaders/csv_loader.py (hunk 1 above)
Comment (DawoudSheraz) [reply course_discovery/apps/course_metadata/data_loaders/csv_loader.py:42]: Here, we are referring to all required fields for ExEd courses. Using AdditionalMetadata word with required fields might cause confusion.

**Review by uzairr — APPROVED** [review]
LGTM 👍

**Pull request merged**